File: 0326-power-of-three/0326-power-of-three.py
class Solution:
    def isPowerOfThree(self, n: int) -> bool:
        return self.power(n)
    
    def power(self, n):
        if n == 1:
            return True
        
        elif n % 3 == 0 and n != 0:
            n /= 3
            return self.power(n)
        return False

File: 0326-power-of-three/0326-power-of-three.java
class Solution {
    public boolean isPowerOfThree(int n) {
        return pow(n);
    }
    static boolean pow(int n){
        if(n==1){
            return true;
        }
        else if(n%3 == 0 && n!=0){
            n=n/3;
            return pow(n);
        }
        return false;
    }
}